### ISSUE TITLE
Develop and Train Breast Cancer Detection Model - Testing001 - Version001 - Model001

### ISSUE THREAD

**Issue by ShenalSen** [ShenalSen/OncoDetect#9]
## Develop and Train Breast Cancer Detection Model Using EfficientNet  

**Description:**  
This issue focuses on implementing a machine learning model for breast cancer detection using mammogram images. The model will leverage the EfficientNet architecture for high accuracy and computational efficiency. Below are the steps to complete this task:  

#### **Step 1: Install Required Libraries**  
- Install necessary libraries for model development and dataset handling.  
- Command:  
  ```bash  
  pip install tensorflow keras numpy pandas scikit-learn matplotlib opencv-python  
  ```  

#### **Step 2: Import Libraries**  
- Import essential libraries for model development and data processing:  
  ```python  
  import numpy as np  
  import matplotlib.pyplot as plt  
  from tensorflow.keras.applications import EfficientNetB0  
  from tensorflow.keras.preprocessing.image import ImageDataGenerator  
  from tensorflow.keras.layers import Dense, Flatten, Dropout  
  from tensorflow.keras.models import Model  
  from tensorflow.keras.optimizers import Adam  
  ```  

#### **Step 3: Load and Configure the Pretrained Model**  
- Use EfficientNetB0 as the base model and add custom layers for binary classification.  
- Compile the model with `adam` optimizer, binary cross-entropy loss, and evaluation metrics (`accuracy`, `AUC`).  

#### **Step 4: Prepare the Dataset**  
- Use `ImageDataGenerator` for preprocessing and augmenting the CBIS-DDSM dataset.  
- Include augmentations like rotation, flipping, and zooming for better generalization.  
- Set up training and validation data loaders.  

#### **Step 5: Train the Model**  
- Train the model for 10 epochs and monitor training and validation accuracy/AUC.  
- Steps per epoch should align with the dataset size and batch size.  

#### **Step 6: Evaluate and Save the Model**  
- Evaluate the model on the validation dataset for loss, accuracy, and AUC.  
- Save the trained model to a file named `oncodetect_model.h5`.  

#### **Step 7: Visualize Results**  
- Plot training and validation metrics (accuracy and AUC) over epochs for performance visualization.  

### **Acceptance Criteria**  
- The model is successfully trained and saved as `oncodetect_model.h5`.  
- Performance metrics (accuracy and AUC) meet expectations for the validation dataset.  
- Plots for training/validation accuracy and AUC are generated and attached to the issue.  

### **Notes**  
- Replace `'path_to_cbis_ddsm'` with the actual dataset path during implementation.  
- Ensure the dataset is preprocessed correctly and all dependencies are installed.
